sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/compatibility/ClassNameResolverPredicate.java

```diff
@@ -0,0 +1,17 @@
+package com.azure.spring.cloud.autoconfigure.implementation.compatibility;
+
+public class ClassNameResolverPredicate {
+
+    boolean resolve(String fullyQuallifiedClassName) {
+        try {
+            if (fullyQuallifiedClassName == null) {
+                return false;
+            }
+
+            Class.forName(fullyQuallifiedClassName);
+            return true;
+        } catch (ClassNotFoundException ex) {
+            return false;
+        }
+    }
+}
```

Comment:
why can't this be a inner class?

Reply:
We will construct an instance of it in the configuration class.  And this makes it easy to inject from ctor, which also help it to be testable.